src/commonMain/kotlin/fr/acinq/lightning/channel/states/Normal.kt

```diff
@@ -874,26 +874,22 @@ data class Normal(
             action.fundingTx.signedTx?.let { add(ChannelAction.Blockchain.PublishTx(it, ChannelAction.Blockchain.PublishTx.Type.FundingTx)) }
             add(ChannelAction.Blockchain.SendWatch(watchConfirmed))
             add(ChannelAction.Message.Send(action.localSigs))
-            // If we received or sent funds as part of the splice, we will add a corresponding entry to our incoming/outgoing payments db
-            addAll(origins.map { origin ->
-                ChannelAction.Storage.StoreIncomingPayment.ViaSpliceIn(
-                    amountReceived = origin.amountReceived(),
-                    serviceFee = origin.fees.serviceFee.toMilliSatoshi(),
-                    miningFee = origin.fees.miningFee,
-                    localInputs = action.fundingTx.sharedTx.tx.localInputs.map { it.outPoint }.toSet(),
-                    txId = action.fundingTx.txId,
-                    origin = origin
-                )
-            })
-            // If we added some funds ourselves it's a swap-in
+            // If we purchased liquidity as part of the channel creation, we will add it to our payments db
+            liquidityPurchase?.let { purchase ->
+                // We only count fees corresponding to the liquidity purchase (i.e. their inputs).
+                add(ChannelAction.Storage.StoreOutgoingPayment.ViaInboundLiquidityRequest(txId = action.fundingTx.txId, purchase = purchase))
+                add(ChannelAction.EmitEvent(LiquidityEvents.Purchased(purchase)))
+            }
+            // NB: the following assume that there can't be a splice-in and a splice-out simultaneously, because we attribute
+            // all local mining fees to both.
             if (action.fundingTx.sharedTx.tx.localInputs.isNotEmpty()) add(
                 ChannelAction.Storage.StoreIncomingPayment.ViaSpliceIn(
                     amountReceived = action.fundingTx.sharedTx.tx.localInputs.map { i -> i.txOut.amount }.sum().toMilliSatoshi() - action.fundingTx.sharedTx.tx.localFees,
                     serviceFee = 0.msat,
                     miningFee = action.fundingTx.sharedTx.tx.localFees.truncateToSatoshi(),
                     localInputs = action.fundingTx.sharedTx.tx.localInputs.map { it.outPoint }.toSet(),
                     txId = action.fundingTx.txId,
-                    origin = null
+                    origin = origins.filterIsInstance<Origin.OnChainWallet>().firstOrNull()
```

Comment:
The model allows multiple origins but I don't think we can have several of the `OnChainWallet` type. We must have an origin otherwise the event will not be stored in db due to https://github.com/ACINQ/lightning-kmp/blob/9bfdfe61e2634c7b1d225fe6a0d09c6622dd1d48/src/commonMain/kotlin/fr/acinq/lightning/payment/IncomingPaymentHandler.kt#L148.

Note that I have a full rework of the incoming payments db on another branch where this particular part is cleaned up.